Mobile: Update gradle, biometric error messages and fix modal issue

cvarley100 opened the pull request:
# Description

- Adds more verbose Biometric error messages
- Fixes biometric modal issue

Fixes # (issue)

## Type of change

- Bug fix (a non-breaking change which fixes an issue)
- Enhancement (a non-breaking change which adds functionality)
- Updates gradle

# How Has This Been Tested?

Tested on device

# Checklist:

_Please delete items that are not relevant._

- [x] My code follows the style guidelines for this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] New and existing unit tests pass locally with my changes
- [x] For changes to `mobile` that include native code (including React Native modules): I have verified that both iOS and Android successfully build in both `Debug` and `Release` modes

cvarley100 committed:
Mobile: Add more verbose biometriometric errors and fix related modal issues

laumair commented:
Why are we removing `src/mobile/android/app/src/main/assets/index.android.bundle`?

cvarley100 commented:
It shouldn't be committed in the first place. We need to add it to gitignore.

cvarley100 commented:
> Why are we removing `src/mobile/android/app/src/main/assets/index.android.bundle`?

It shouldn't be committed in the first place. We need to add it to gitignore.

rajivshah3 commented:
I added `index.android.bundle` to the gitignore in #968. It might cause a merge conflict here, though

cvarley100 committed:
Mobile: Fix modal rerendering (#959)

laumair commented:
@cvarley100 Merge conflicts.

rajivshah3 committed:
Android build fixes (#969)

* Mobile/Android: Use 'bin' version of Gradle

* Mobile/Android: Use build-tools 28.0.3 to suppress warning

* Mobile/Android: Enable aapt2 to suppress warning

* Mobile/Android: Remove drawable-mdpi folder as it will cause errors

When building, these assets are automatically generated. Checking them into this folder will cause errors with duplicate resources

rajivshah3 committed:
Merge branch 'develop' into bug/biometric

laumair commented on src/mobile/src/ui/components/EnterPassword.js:
```diff
@@ -103,7 +105,9 @@ class EnterPassword extends Component {
     hideModal() {
-        this.props.toggleModalActivity();
+        if (this.props.isModalActive) {
+            this.props.toggleModalActivity();
```
What happens if `toggleModalActivity` is dispatched when modal is already inactive?

cvarley100 replied:
Well this function is just for hiding the modal.

laumair replied:
Oh didn't realise it was for just hiding modal as it also [accepts](https://github.com/iotaledger/trinity-wallet/blob/develop/src/shared/actions/ui.js#L217) `modalProps` and `modalContent`. 

Perhaps the condition (if modal is active) is moved to the redux action itself. 

```
const hideModal = () => {
    return (dispatch, getState) => {
        if (getState().ui.isModalActive) { dispatch(toggleModalActivity()); };
    };
};
```

That way we won't need to include `isModalActive` prop in every component where we need to close the modal.

cvarley100 replied:
This was more as a failsafe. I don't think they are necessary however - I will remove them.

cvarley100 replied:
FYI, `toggleModalActivity` has two purposes - opening the right modal with the right props. But without props it simply closes an open modal.

cvarley100 committed:
Mobile: Remove unnecessary hideModal checks
